File: src/main/java/com/rekeningrijden/simulation/entities/SubRoute.java
package com.rekeningrijden.simulation.entities;

import java.util.ArrayList;
import java.util.List;

public class SubRoute {
    private String countryCode; //Country the car is driving in
    private List<Coordinate> coordinates;
    private boolean subRouteDriven;

    public SubRoute(String countryCode) {
        coordinates = new ArrayList<>();
        this.countryCode = countryCode;
        this.subRouteDriven = false;
    }

    public List<Coordinate> getCoordinates() {
        return coordinates;
    }

    public boolean isSubRouteDriven() {
        return subRouteDriven;
    }

    public void setSubRouteDriven(boolean subRouteDriven) {
        this.subRouteDriven = subRouteDriven;
    }

    public void addCoordinate(Coordinate coor) {
        this.coordinates.add(coor);
    }

    public String getCountryCode() {
        return countryCode;
    }

    public Coordinate getNextCoordinateAtIndex(int index){
        if (index >= coordinates.size()){
            this.subRouteDriven = true;
            return null;
        }
        return coordinates.get(index);
    }
}

File: src/main/kotlin/com/rekeningrijden/simulation/entities/SubRoute.kt
package com.rekeningrijden.simulation.entities

import com.rekeningrijden.simulation.coordinate.CoordinateImpl

data class SubRoute(
    /**
     * The two-letter ISO code indicating what country the car is currently
     * driving in.
     */
    val countryCode: String
) {
    val coordinates = mutableListOf<CoordinateImpl>()
    var isSubRouteDriven: Boolean = false

    fun getNextCoordinateAtIndex(index: Int): CoordinateImpl? {
        if (index >= coordinates.size) {
            this.isSubRouteDriven = true
            return null
        }
        return coordinates[index]
    }
}
